App/Commands/BaseCommandController.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace App\Commands;
+
+use App\Console\InputConsole;
+use App\Console\InputValidator;
+use App\Console\OutputConsole;
+use App\Enums\FastType;
+use App\Model\Fast;
+use App\Store\StoreManager;
+use Exception;
+
+class BaseCommandController
+{
+
+    protected array $fastTypes;
+    protected array $confirmationOptions = [
+        "Y" => true,
+        "N" => false
+    ];
+
+    public function __construct(
+        protected InputConsole   $input,
+        protected OutputConsole  $output,
+        protected StoreManager   $store,
+        protected InputValidator $validator,
+        protected Fast           $newFast
+    )
+    {
+        $this->setFastTypes();
+    }
+
+    /**
+     * Sets fast starting date from user input.
+     * @throws Exception
+     */
+    protected function getStartDate()
+    {
+        $this->output->writeYellow('Enter Start Date of Fast format:(Y-m-d H:i:s) => (2020-10-10 20:00:00)');
```

Comment:
I would suggest refactoring like this to make it more usable

$this->output->write('text', COLORS.YELLOW)

or

$this->output->write('text', options)